### PR TITLE
Removing outdated flow.run() docs

### PR DESCRIPTION
**Pull request by atravitz**
## Description
I'm migrating the "operations and conditions" section into the "the flow project" section to remove redundancies and outdated info.

## Motivation and Context
addresses [flow issue #513] (https://github.com/glotzerlab/signac-flow/issues/513)

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

File: docs/source/flow-project.rst [hunk 1]
```diff
@@ -11,22 +11,22 @@ This chapter describes how to setup a complete workflow via the implementation o
...
         pass
 
     if __name__ == '__main__':
-        Project().main()
+        MyProject().main()
 
 .. tip::
 
-    You can generate boiler-plate templates like the one above with the ``$ flow init`` function.
+    The ``$ flow init`` function will generate a boiler-plate ``project.py`` like the one above.
```
Comment (bdice) [inline docs/source/flow-project.rst:29]:
```suggestion
    The ``$ flow init`` function will generate a minimal ``project.py`` file like the one above.
```

File: docs/source/flow-project.rst [hunk 2]
```diff
@@ -48,8 +48,68 @@ Executing this script on the command line will give us access to this project's
...
-We will reproduce the simple workflow introduced in the previous section by first copying both the ``greeted()`` condition function and the ``hello()`` *operation* function into the ``project.py`` module.
-We then use the :py:meth:`~flow.FlowProject.operation` and the :py:meth:`~flow.FlowProject.post` decorator functions to specify that the ``hello()`` operation function is part of our workflow and that it should only be executed if the ``greeted()`` condition is not met.
+This chapter introduces the two **fundamental concepts** for the implementation of workflows with the **signac-flow** package: :ref:`operations` and :ref:`conditions`.
+
+
+.. _operations:
+
+It is highly recommended to divide individual modifications of your project's data space into distinct functions.
+
+In this context, an *operation* is defined as a unary function where its only argument is aninstance of :py:class:`~signac.contrib.job.Job`.
```
Comment (bdice) [inline docs/source/flow-project.rst:58]:
```suggestion
In this context, an *operation* is defined as a function whose only positional argument is an instance of :py:class:`~signac.contrib.job.Job` (in the special case of :ref:`aggregate operations <aggregation>`, variable positional arguments ``*jobs`` are permitted).
```

File: docs/source/flow-project.rst [hunk 3]
```diff
@@ -48,8 +48,68 @@ Executing this script on the command line will give us access to this project's
...
+            with open('hello.txt', 'w') as file:
+                file.write('world!\n')
+
+    if __name__ == '__main__':
+        Project().main()
+
+
+.. _conditions:
+
+Here the :py:meth:`~flow.FlowProject.operation` decorator function specifies that the ``hello()`` operation function is part of our workflow. If we run ``python project.py run``, signac-flow will execute all ``hello()`` for all jobs in the project.
```
Comment (bdice) [inline docs/source/flow-project.rst:98]:
```suggestion
Here the :py:meth:`~flow.FlowProject.operation` decorator function specifies that the ``hello`` operation function is part of our workflow. If we run ``python project.py run``, **signac-flow** will execute ``hello`` for all jobs in the project.
```

File: docs/source/flow-project.rst [hunk 4]
```diff
@@ -11,22 +11,22 @@ This chapter describes how to setup a complete workflow via the implementation o
 Setup and Interface
 ===================
 
-To implement a more automated workflow, we can subclass a :py:class:`~flow.FlowProject`:
+To implement an automated workflow using **signac flow**, we create a subclass of :py:class:`~flow.FlowProject`, here named :py:class:`~MyProject`:
```
Comment (bdice) [inline docs/source/flow-project.rst:14]:
```suggestion
To implement an automated workflow using **signac-flow**, we create a subclass of :py:class:`~flow.FlowProject`, here named ``MyProject``:
```

File: docs/source/flow-project.rst [hunk 5]
```diff
@@ -48,8 +48,68 @@ Executing this script on the command line will give us access to this project's
...
+
+    if __name__ == '__main__':
+        Project().main()
+
+
+.. _conditions:
+
+Here the :py:meth:`~flow.FlowProject.operation` decorator function specifies that the ``hello()`` operation function is part of our workflow. If we run ``python project.py run``, signac-flow will execute all ``hello()`` for all jobs in the project.
+
+However, we only want to perform ``hello()`` if ``hello.txt`` does not yet exist in the ``job``'s workspace. To do this, need to create a condition function named ``greeted()`` that tells us if ``hello.txt`` already exits in the job workspace:
```
Comment (bdice) [inline docs/source/flow-project.rst:100]:
```suggestion
However, we only want to execute ``hello`` if ``hello.txt`` does not yet exist in the job's workspace.
To do this, we need to create a condition function named ``greeted`` that tells us if ``hello.txt`` already exists in the job workspace:
```

File: docs/source/flow-project.rst [hunk 6]
```diff
@@ -48,8 +48,68 @@ Executing this script on the command line will give us access to this project's
...
+
+.. code-block:: python
+    # project.py
+
+    ...
+
+    def greeted(job):
+        return job.isfile('hello.txt')
+
+To complete this component of the workflow, we use the :py:meth:`~flow.FlowProject.post` decorator function to specify that the ``hello()`` operation function should only be executed if the ``greeted()`` condition is not met.
```
Comment (bdice) [inline docs/source/flow-project.rst:110]: Let's remove parentheses that indicate calling a function unless including the arguments like `greeted(job)`.
```suggestion
To complete this component of the workflow, we use the :py:meth:`~flow.FlowProject.post` decorator function to specify that the ``hello`` operation function should only be executed if the ``greeted`` condition is not met.
```

**Review by bdice — COMMENTED** [review]
Some preliminary suggestions. Thanks for starting work on this @atravitz!

File: docs/source/flow-project.rst (hunk 2 above)
Comment (atravitz) [reply docs/source/flow-project.rst:58]: helpful clarification, thanks!

File: docs/source/flow-project.rst [hunk 7]
```diff
@@ -4,29 +4,29 @@
 The FlowProject
 ===============
 
-This chapter describes how to setup a complete workflow via the implementation of a :py:class:`~flow.FlowProject`.
+This chapter describes how to setup a complete workflow via the implementation of a :py:class:`~flow.FlowProject`. It includes two **fundamental concepts** for the implementation of workflows with the **signac-flow** package: :ref:`operations <operations>` and :ref:`conditions <conditions>`.
```
Comment (bdice) [inline docs/source/flow-project.rst:7]:
```suggestion
This chapter describes how to setup a complete workflow via the implementation of a :py:class:`~flow.FlowProject`.
It includes two **fundamental concepts** for the implementation of workflows with the **signac-flow** package: :ref:`operations <operations>` and :ref:`conditions <conditions>`.
```

File: docs/source/flow-project.rst [hunk 8]
```diff
@@ -39,44 +39,104 @@ Executing this script on the command line will give us access to this project's
...
 
-.. _workflow-definition:
+.. _operations:
 
-Defining a workflow
-===================
+Operations
+==========
+
+
```
Comment (bdice) [inline docs/source/flow-project.rst:51]: Remove extra newline.
```suggestion
```

File: docs/source/flow-project.rst [hunk 9]
```diff
@@ -39,44 +39,104 @@ Executing this script on the command line will give us access to this project's
...
             with open('hello.txt', 'w') as file:
                 file.write('world!\n')
 
 
     if __name__ == '__main__':
-        Project().main()
+        MyProject().main()
 
-We can define both *pre* and *post* conditions, which allow us to define arbitrary workflows as an acyclic graph.
+We can define both :py:meth:`~flow.FlowProject.pre` and :py:meth:`~flow.FlowProject.post` conditions, which allow us to define arbitrary workflows as an acyclic graph.
```
Comment (bdice) [inline docs/source/flow-project.rst:139]: We should link to a definition if we use "directed acyclic graph" as a math/CS term.
```suggestion
We can define both :py:meth:`~flow.FlowProject.pre` and :py:meth:`~flow.FlowProject.post` conditions, which allow us to define arbitrary workflows as a `directed acyclic graph <https://en.wikipedia.org/wiki/Directed_acyclic_graph>`__.
```

File: docs/source/flow-project.rst [hunk 10]
```diff
@@ -109,7 +169,7 @@ We can then execute this workflow with:
     hello 2b985fa90138327bef586f9ad87fc310
     # ...
 
-If we implemented and integrated the operation and condition functions correctly, calling the ``run`` command twice should produce no output the second time, since the ``greeted()`` condition is met for all jobs and the ``hello()`` operation should therefore not be executed.
+If we implemented and integrated the operation and condition functions correctly, calling the ``run`` command twice should produce no output the second time, since the ``greeted`` condition is met for all jobs and the ``hello`` operation should therefore not be executed.
```
Comment (bdice) [inline docs/source/flow-project.rst:172]: Calling `run` again might print progress output or something like that, but we can promise that it will "take no action" or "execute no operations."
```suggestion
If we implemented and integrated the operation and condition functions correctly, calling the ``run`` command twice should execute no operations the second time, since the ``greeted`` condition is met for all jobs and the ``hello`` operation should therefore not be executed.
```

**Review by bdice — COMMENTED** [review]
This looks great! A couple minor suggestions, I'll leave the rest to other reviewers.

**Review by lyrivera — COMMENTED** [review]
Not sure about the formatting standards for `signac` but found a few extra lines between sentences and also between text and code-blocks. I didn't point out the double blank lines within code-blocks. LGTM otherwise.

File: docs/source/flow-project.rst (hunk 7 above)
Comment (cbkerr) [reply docs/source/flow-project.rst:7]: I think as long as operations and conditions appear as sub-entries in the sidebar toc, we don't need to announce them in the organizational statement (the second sentence).

This sentence makes it sound like there is a central list of "fundamental concepts", which is perhaps another motivation for having a glossary.

File: docs/source/flow-project.rst [hunk 11]
```diff
@@ -39,44 +39,103 @@ Executing this script on the command line will give us access to this project's
 
 .. note::
 
-    You can have **multiple** implementations of :py:class:`~flow.FlowProject` that all operate on the same **signac** project!
+    You can have **multiple** implementations of :py:class:`~flow.FlowProject` that all operate on the same **signac** data space!
     This may be useful, for example, if you want to implement two very distinct workflows that operate on the same data space.
     Simply put those in different modules, *e.g.*, ``project_a.py`` and ``project_b.py``.
 
-.. _workflow-definition:
```
Comment (cbkerr) [inline docs/source/flow-project.rst:46]: Don't have time to check rn if this was used before/if it needs to be added to a redirect list or anything for any people's saved URLs.

**Review by cbkerr — CHANGES_REQUESTED** [review]
The changes to the files (and removing old recommendations) are good and I agree that these sections should be together, but I'm concerned with removing a key section from the sidebar.

New users exploring the docs intuitively know about "operations" and "conditions" but don't know what to expect from FlowProject. Users who go there by muscle memory might not know to look under FlowProject.

**Proposal**: Change whatever renders the toc entry to say "Workflow Setup" or maybe "FlowProject for Workflow"

File: docs/source/flow-project.rst (hunk 11 above)
Comment (cbkerr) [reply docs/source/flow-project.rst:46]: Currently the url that links to the Defining a workflow section. 

https://docs.signac.io/en/latest/flow-project.html#defining-a-workflow

Perhaps the right course of action is to leave both targets for this section?

**Review by cbkerr — APPROVED** [review]
Looks good, and sidebar naming that I brought up is out of scope for this PR

**Review by lyrivera — APPROVED** [review]
LGTM

File: docs/source/flow-project.rst [hunk 12]
```diff
@@ -5,28 +5,29 @@ The FlowProject
...
         pass
 
     if __name__ == '__main__':
-        Project().main()
+        MyProject().main()
 
 .. tip::
 
-    You can generate boiler-plate templates like the one above with the ``$ flow init`` function.
+    The ``$ flow init`` function will generate a minimal ``project.py`` file like the one above.
```
Comment (bdice) [inline docs/source/flow-project.rst:30]:
```suggestion
    The ``$ flow init`` command will generate a minimal ``project.py`` file like the one above.
```

File: docs/source/flow-project.rst [hunk 13]
```diff
@@ -153,20 +211,15 @@ The Project Status
 The :py:class:`~flow.FlowProject` class allows us to generate a **status** view of our project.
 The status view provides information about which conditions are met and what operations are pending execution.
 
-A condition function which is supposed to be shown in the **status** view is called a *label-function*.
+A condition function which will be shown in the **status** view is called a *label-function*.
```
Comment (bdice) [inline docs/source/flow-project.rst:214]: Invert sentence order so that it's clear that we're defining a new term.
```suggestion
A *label-function* is a condition function which will be shown in the **status** view.
```

File: docs/source/flow-project.rst [hunk 14]
```diff
@@ -215,7 +268,7 @@ As shown before, all *eligible* operations can then be executed with:
 
     ~/my_project $ python project.py run
 
-The status determination operates in serial by default, because typically the overhead costs of using threads/processes are large. However this can be configured by setting a value for the ``flow.status_parallelization`` configuration key.
+The status determination operates in series by default, because typically the overhead costs of using threads/processes are large. However this can be configured by setting a value for the ``flow.status_parallelization`` configuration key.
```
Comment (bdice) [inline docs/source/flow-project.rst:271]: I'm not sure if this change is correct. To the best of my knowledge, "serial execution" is a term, but "series execution" is not a term.

**Review by bdice — CHANGES_REQUESTED** [review]
@atravitz I have 3 small change requests, if you agree with those then apply them and merge this PR. Great job!

File: docs/source/flow-project.rst (hunk 14 above)
Comment (cbkerr) [reply docs/source/flow-project.rst:271]: Bradley and I discussed offline. We agreed that series is the noun form of serial but that the adjective form seems to be the norm in the language of algorithms. Referencing Wikipedia and the OED (serial and sequential 2b), we agreed to switch to "Status is determined sequentially..."

Wikipedia:
>Concurrency and parallelism are in general distinct concepts, but they often overlap – many distributed algorithms are both concurrent and parallel – and thus "sequential" is used to contrast with both, without distinguishing which one. If these need to be distinguished, the opposing pairs sequential/concurrent and serial/parallel may be used.

OED:
> seqential def 2b: Of, pertaining to, or designating various aspects of a computer system and its control programming that operate or are utilized serially; "

Comment (cbkerr) [inline docs/source/flow-project.rst:271]:
```suggestion
Status is determined sequentially because typically the overhead costs of using threads/processes are large. However, this can be configured by setting a value for the ``flow.status_parallelization`` configuration key.
```

Comment (cbkerr) [reply docs/source/flow-project.rst:271]:
```suggestion
Status is determined sequentially by default, because typically the overhead costs of using threads/processes are large. However, this can be configured by setting a value for the ``flow.status_parallelization`` configuration key.
```

File: docs/source/flow-project.rst [hunk 15]
```diff
@@ -215,7 +268,7 @@ As shown before, all *eligible* operations can then be executed with:
 
     ~/my_project $ python project.py run
 
-The status determination operates in serial by default, because typically the overhead costs of using threads/processes are large. However this can be configured by setting a value for the ``flow.status_parallelization`` configuration key.
+Status is determined sequentially by default, because typically the overhead costs of using threads/processes are large. However, this can be configured by setting a value for the ``flow.status_parallelization`` configuration key.
```
Comment (bdice) [inline docs/source/flow-project.rst:271]: Break sentences across lines.
```suggestion
Status is determined sequentially by default, because typically the overhead costs of using threads/processes are large.
However, this can be configured by setting a value for the ``flow.status_parallelization`` configuration key.
```

File: docs/source/flow-project.rst [hunk 16]
```diff
@@ -39,44 +40,101 @@ Executing this script on the command line will give us access to this project's
...
+        MyProject().main()
+
+
+.. _conditions:
+
+Conditions
+==========
 
-    class Project(FlowProject):
+Here the :py:meth:`~flow.FlowProject.operation` decorator function specifies that the ``hello`` operation function is part of our workflow. If we run ``python project.py run``, **signac-flow** will execute ``hello`` for all jobs in the project.
```
Comment (bdice) [inline docs/source/flow-project.rst:97]:
```suggestion
Here the :py:meth:`~flow.FlowProject.operation` decorator function specifies that the ``hello`` operation function is part of our workflow.
If we run ``python project.py run``, **signac-flow** will execute ``hello`` for all jobs in the project.
```

**Review by bdice — APPROVED** [review]
Merging.
